server/base/src/main/java/org/apache/accumulo/server/fs/SpaceAwareVolumeChooser.java

```diff
@@ -72,9 +67,7 @@ public String choose(VolumeChooserEnvironment env, String[] options)
       VolumeChooserEnvironment env) {
 
     if (choiceCache == null) {
-      ServerConfigurationFactory scf = loadConfFactory(env);
-      AccumuloConfiguration systemConfiguration = scf.getSystemConfiguration();
-      String propertyValue = systemConfiguration.get(HDFS_SPACE_RECOMPUTE_INTERVAL);
+      String propertyValue = env.getContext().getConfiguration().getCustom(RECOMPUTE_INTERVAL);
```

Comment:
> I think a better implementation is to stop requiring properties to be prefixed with "general.custom" and allowing them to have any arbitrary property name in our config.

That sounds like a fundamental change to Accumulo which could be done in a later PR.  This change is exposing current functionality to plugins in an easy to use manner.

Reply:
> This is convenient API, but I worry that this obscures the configuration properties outside of code.

My intent was to address a problem I saw with every plugin using these prefixes.  They all wrote the same boilerplate code in slightly different ways which made the plugins harder to understand and read.  The methods have very clear javadoc.  Do you have any suggestions for avoiding the boilerplate code that would make code more readable?

Reply:
> That sounds like a fundamental change to Accumulo which could be done in a later PR.

I agree. I just didn't know if that might change your interim solution in any way.

> Do you have any suggestions for avoiding the boilerplate code that would make code more readable?

I think the main thing is that we just need to make it clear what the full property name actually is, so that there's no confusion when the user tries to set/retrieve it using the shell, or in the accumulo-site config file, or in their own configuration management tooling. This could be done by ensuring the full property name is available in a constant in the class, or clearly marked in javadoc. Any constants which use the shorter suffix should be hidden from the user in a private fields and shouldn't be returned or requested in non-private methods (other than the convenience method for prefixing and retrieving from config, of course).

Reply:
> I think the main thing is that we just need to make it clear what the full property name actually is

The current javadoc for these methods does this.